### PR TITLE
Add unit tests for `DirectFund.Crank()`

### LINKED ISSUE #51: Add state transition tests to `DirectFundingObjectiveState.` `Update`, `Crank`

### PR DESCRIPTION
**Pull request by geoknee**
Brings test coverage of the `protocols` package up to ~86%.

Does not yet assert on the correct `SideEffects` -- but those are sill TODO anyway #53 .

Closes #51.

### CODE REVIEW

File: protocols/direct-funding_test.go [hunk 1]
```diff
@@ -96,3 +96,77 @@ func TestUpdate(t *testing.T) {
...
+		t.Error(err)
+	}
+	if waitingFor != WaitingForCompletePostFund {
+		t.Errorf(`WaitingFor: expected %v, got %v`, WaitingForCompletePostFund, waitingFor)
+	}
+
+	// Manually progress the extended state by collecting postfund signatures
+	o.(DirectFundingObjectiveState).PostFundSigned[0] = true
+	o.(DirectFundingObjectiveState).PostFundSigned[1] = true
+	o.(DirectFundingObjectiveState).PostFundSigned[2] = true
```
Comment (NiloCK) [inline protocols/direct-funding_test.go:159]: There's a comment to be made here that `PostFundSigned` should probably be `postFundSigned` - that updates to, especially, the security-related state should not be possible from outside of each individual `Objective` implementation.

That said, you've done it here only by use of the explicit cast, which we don't intend to employ outside of test code.

No suggestion - just flagging something to think about.

**Review by NiloCK — APPROVED** [review]
:+1:

File: protocols/direct-funding_test.go (hunk 1 above)
Comment (geoknee) [reply protocols/direct-funding_test.go:159]: I agree. Let's make this change in a rapid, follow up PR (it will set a precedent for the other protocols).

Comment (geoknee) [reply protocols/direct-funding_test.go:159]: #84

**Pull request merged**